packages/apps/test/commands/domains/add.test.ts

```diff
@@ -17,14 +19,96 @@ describe('domains:add', () => {
     status: 'pending'
   }
 
-  test
-    .stderr()
-    .nock('https://api.heroku.com', api => api
-      .post('/apps/myapp/domains', {hostname: 'example.com'})
-      .reply(200, domainsResponse)
-    )
-    .command(['domains:add', 'example.com', '--app', 'myapp'])
-    .it('adds the domain to the app', ctx => {
-      expect(ctx.stderr).to.contain('Adding example.com to myapp... done')
+  describe('adding a domain without the feature flag on (the old way)', () => {
+    test
+      .stderr()
+      .nock('https://api.heroku.com', api => api
+        .get('/apps/myapp/features/allow-multiple-sni-endpoints')
+        .reply(200, {
+          enabled: false
+        })
+        .post('/apps/myapp/domains', {hostname: 'example.com'})
+        .reply(200, domainsResponse)
+      )
+      .command(['domains:add', 'example.com', '--app', 'myapp'])
+      .it('adds the domain to the app', ctx => {
+        expect(ctx.stderr).to.contain('Adding example.com to myapp... done')
+      })
+  })
+
+  describe('adding a domain to an app with multiple certs', () => {
+    describe('using the --cert flag', () => {
+      test
+        .stderr()
+        .nock('https://api.heroku.com', api => api
+          .get('/apps/myapp/features/allow-multiple-sni-endpoints')
+          .reply(200, {
+            enabled: true
+          })
+          .post('/apps/myapp/domains', {
+            hostname: 'example.com',
+            sni_endpoint: 'my-cert'
+          })
+          .reply(200, domainsResponse)
+        )
+        .command(['domains:add', 'example.com', '--app', 'myapp', '--cert', 'my-cert'])
+        .it('adds the domain to the app', ctx => {
+          expect(ctx.stderr).to.contain('Adding example.com to myapp... done')
+        })
+    })
+
+    describe('without passing a cert', () => {
+      const certsResponse = [
+        {
+          app: {
+            name: 'myapp',
+          },
+          name: 'cert1',
+          displayName: 'Best Cert Ever',
+          ssl_cert: {
+            cert_domains: ['foo.com', 'bar.com', 'baz.com', 'baq.com', 'blah.com', 'rejairieja.com'],
+          },
+        },
+        {
+          app: {
+            name: 'myapp',
+          },
+          name: 'cert2',
+          ssl_cert: {
+            cert_domains: ['foo.com', 'bar.com', 'baz.com', 'baq.com', 'blah.com', 'rejairieja.com'],
+          },
+        },
+      ]
+
+      test
+        .stderr()
+        .stub(inquirer, 'prompt', () => {
+          return Promise.resolve({cert: 'my-cert'})
+        })
+        .nock('https://api.heroku.com', api => api
+          .get('/apps/myapp/features/allow-multiple-sni-endpoints')
+          .reply(200, {
+            enabled: true
+          })
+          .post('/apps/myapp/domains', {
+            hostname: 'example.com',
+          })
+          .reply(422, {
+            id: 'invalid_params',
+            message: '\'sni_endpoint\' param is required when adding a domain to an app with multiple SSL certs.'
```

Comment:
I think this is fine for now, but I do find an issue with sni_endpoint... I really wish we just call it --cert or something